### PR TITLE
adds support to display passed rules

### PR DESCRIPTION
**Pull request by patilpankaj212**
Added a new flag `--show-passed`, which when used, would display passed rules along with violations.

### CODE REVIEW

File: pkg/cli/scan.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func init() {
 	scanCmd.Flags().BoolVarP(&scanOptions.configOnly, "config-only", "", false, "will output resource config (should only be used for debugging purposes)")
 	// flag passes a string, but we normalize to bool in PreRun
 	scanCmd.Flags().StringVar(&scanOptions.useColors, "use-colors", "auto", "color output (auto, t, f)")
-	scanCmd.Flags().BoolVarP(&scanOptions.Verbose, "verbose", "v", false, "will show violations with details (applicable for default output)")
+	scanCmd.Flags().BoolVarP(&scanOptions.Verbose, "verbose", "v", false, "will show violations with details for human output and passed tests for junit-xml")
```
Comment (dev-gaur) [inline pkg/cli/scan.go:65]: Do we need this here in flag description? we can just add this "passed tests for junit-xml" in docs.

File: pkg/results/types.go [hunk 2]
```diff
@@ -50,26 +60,30 @@ type ScanSummary struct {
 	IacType              string `json:"iac_type" yaml:"iac_type" xml:"iac_type,attr"`
 	Timestamp            string `json:"scanned_at" yaml:"scanned_at" xml:"scanned_at,attr"`
 	ShowViolationDetails bool   `json:"-" yaml:"-" xml:"-"`
-	TotalPolicies        int    `json:"policies_validated" yaml:"policies_validated" xml:"policies_validated,attr"`
-	ViolatedPolicies     int    `json:"violated_policies" yaml:"violated_policies" xml:"violated_policies,attr"`
+	RulesValidated       int    `json:"rules_validated" yaml:"rules_validated" xml:"rules_validated,attr"`
```
Comment (dev-gaur) [inline pkg/results/types.go:63]: The work `RuleValidated` gives the same sound as rules approved. Could we go with Rules Tried?

File: pkg/writer/junit_xml_test.go [hunk 3]
```diff
@@ -85,14 +85,15 @@ func TestJUnitXMLWriter(t *testing.T) {
...
-							IacType:          "k8s",
-							Timestamp:        "2020-12-12 11:21:29.902796 +0000 UTC",
-							TotalPolicies:    550,
-							LowCount:         0,
-							MediumCount:      0,
-							HighCount:        1,
-							ViolatedPolicies: 1,
+							ResourcePath:    "test_resource_path",
+							IacType:         "k8s",
+							Timestamp:       "2020-12-12 11:21:29.902796 +0000 UTC",
```
Comment (dev-gaur) [inline pkg/writer/junit_xml_test.go:90]: If you could not hardcode the date string, that'd be great and holistic. its fine this way for now as well

File: pkg/writer/yaml_test.go [hunk 4]
```diff
@@ -109,8 +110,9 @@ const (
         file/folder: test
         iac_type: terraform
         scanned_at: 2020-12-12 11:21:29.902796 +0000 UTC
-        policies_validated: 566
-        violated_policies: 1
+        rules_validated: 566
+        rules_violated: 1
```
Comment (dev-gaur) [inline pkg/writer/yaml_test.go:114]: What's the difference between rule_violated and total_violations?

**Review by dev-gaur — CHANGES_REQUESTED** [review]
Minor questions asked and changed proposed.

File: pkg/results/types.go (hunk 2 above)
Comment (kanchwala-yusuf) [reply pkg/results/types.go:63]: May be `Evaluated`?

**Review by kanchwala-yusuf — DISMISSED** [review]

**Comment by kanchwala-yusuf** [comment]
Please get the output reviewed once with respect to fields `rules_violated`, `rules_validated` and `total_violations`. May be we can get the output reviewed from a couple of folks and see if it not causing confusions.

File: pkg/http-server/file-scan.go [hunk 5]
```diff
@@ -93,6 +95,18 @@ func (g *APIHandler) scanFile(w http.ResponseWriter, r *http.Request) {
 	// severity is the minimum severity level of violations that the user want to get informed about: low, medium or high
 	severity := r.FormValue("severity")
 
+	// if an error occurs while reading the attribute, we log the error and proceed with default value
+	showPassedValue := r.FormValue("show_passed")
+	if showPassedValue != "" {
+		showPassed, err = strconv.ParseBool(showPassedValue)
+		if err != nil {
+			errMsg := fmt.Sprintf("error while reading 'show_passed' value. error: '%v'", err)
```
Comment (kanchwala-yusuf) [inline pkg/http-server/file-scan.go:103]: May be we can also return the incorrect value of `show_passed` variable

**Review by kanchwala-yusuf — APPROVED** [review]

**Comment by kanchwala-yusuf** [comment]
@patilpankaj212 , there seem to be some conflicts between master and this branch. Please resolve them

**Review by kanchwala-yusuf — APPROVED** [review]

File: test/e2e/help/golden/help_scan.txt [hunk 6]
```diff
@@ -18,6 +18,7 @@ Flags:
   -u, --remote-url string         url pointing to remote IaC repository
       --scan-rules strings        one or more rules to scan (example: --scan-rules="ruleID1,ruleID2")
       --severity string           minimum severity level of the policy violations to be reported by terrascan
+      --show-passed               will display rules that were not triggered
```
Comment (kanchwala-yusuf) [inline test/e2e/help/golden/help_scan.txt:21]: @patilpankaj212 , @amirbenv , I feel the help message here can improve. `will display rules that were not triggered` does not sound right. `--show-passed` and `not triggered` don't really go hand in hand.

May be we can have it as `lists rules that weren't violated` or  simply `display passed rules`? or something that goes along with the flag `--show-passed`.

**Review by kanchwala-yusuf — CHANGES_REQUESTED** [review]

File: test/e2e/help/golden/help_scan.txt (hunk 6 above)
Comment (amirbenv) [reply test/e2e/help/golden/help_scan.txt:21]: How about `display passed rules, along with violations`.

**Review by kanchwala-yusuf — APPROVED** [review]

**Comment by ismailyenigul** [comment]
Hi @kanchwala-yusuf @patilpankaj212 
Thanks for this feature but It reports passes for all controls on terrascan rules.
But I think it should only report related passes which is available in the scanned resources.
In my following case, it has only 2  aws resources 1 data.
resource "aws_sns_topic"
resource "aws_kms_key" 
data "aws_iam_policy_document"

but `--show-passed reports everything(GCP checks, Azure, Kubernetes).
I expect that it should reports checks for sns, kms and iam policy.
I will open a feature request for this if you agree?

Thanks

**Pull request merged**